compress.cc

```diff
@@ -123,11 +146,21 @@ compressor::ptr_type compressor::create(uint32_t id) {
         }
     }
 
-    opt_getter dummy;
+    opt_getter dummy = [](auto&&) {
+        return std::nullopt;
+    };
+
+    static thread_local std::unordered_map<uint32_t, ptr_type> id_compressors;
+
+    auto i = id_compressors.find(id);
+    if (i != id_compressors.end()) {
+        return i->second;
```

Comment:
No need to `replace` here?

Reply:
No. Compressors created via id:s never had any options set, nor do we have any to provide.

Reply:
Please write this in a comment